File: example/examples/overlays/circle.js
import React, { Component } from 'react'
import { StyleSheet } from 'react-native'
import { MapView } from 'react-native-baidumap-sdk'

export default class Circle extends Component {
  static navigationOptions = { title: 'Circle' }

  render() {
    return (
      <MapView style={StyleSheet.absoluteFill}>
        <MapView.Circle
          center={{ latitude: 39.914884, longitude: 116.403883 }}
          radius={10000}
          strokeWidth={2}
          strokeColor="rgba(0, 0, 255, 0.5)"
          fillColor="rgba(255, 0, 0, 0.5)"
        />
      </MapView>
    )
  }
}

File: lib/android/src/main/java/qiuxiang/baidu_map/map_view/Circle.kt
package qiuxiang.baidu_map.map_view

import android.content.Context
import android.graphics.Color
import com.baidu.mapapi.map.BaiduMap
import com.baidu.mapapi.map.Circle
import com.baidu.mapapi.map.CircleOptions
import com.baidu.mapapi.map.Stroke
import com.baidu.mapapi.model.LatLng
import com.facebook.react.views.view.ReactViewGroup

class Circle(context: Context) : ReactViewGroup(context), Overlay {
  private var circle: Circle? = null

  var center: LatLng? = null
    set(value) {
      field = value
      circle?.center = value
    }

  var radius: Int = 0
    set(value) {
      field = value
      circle?.radius = value
    }

  var strokeWidth: Int = 1
    set(value) {
      field = value
      circle?.stroke = Stroke(value, strokeColor)
    }

  var strokeColor: Int = Color.BLACK
    set(value) {
      field = value
      circle?.stroke = Stroke(strokeWidth, value)
    }

  var fillColor: Int = Color.BLACK
    set(value) {
      field = value
      circle?.fillColor = value
    }

  var zIndex: Int = 0
    set(value) {
      field = value
      circle?.zIndex = value
    }

  override fun add(map: BaiduMap) {
    circle = map.addOverlay(
      CircleOptions()
        .center(center)
        .radius(radius)
        .stroke(Stroke(strokeWidth, strokeColor))
        .fillColor(fillColor)
        .zIndex(zIndex)
    ) as Circle
  }

  override fun remove() {
    circle?.remove()
  }
}
